src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseTestCaseService.java

```diff
@@ -0,0 +1,158 @@
+package de.tum.in.www1.artemis.service;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.Feedback;
+import de.tum.in.www1.artemis.domain.ProgrammingExercise;
+import de.tum.in.www1.artemis.domain.ProgrammingExerciseTestCase;
+import de.tum.in.www1.artemis.domain.Result;
+import de.tum.in.www1.artemis.domain.enumeration.FeedbackType;
+import de.tum.in.www1.artemis.repository.ProgrammingExerciseTestCaseRepository;
+
+@Service
+public class ProgrammingExerciseTestCaseService {
+
+    private final ProgrammingExerciseTestCaseRepository testCaseRepository;
+
+    public ProgrammingExerciseTestCaseService(ProgrammingExerciseTestCaseRepository testCaseRepository) {
+        this.testCaseRepository = testCaseRepository;
+    }
+
+    // TODO: Workaround for known bug, should be removed once fixed: https://jira.spring.io/browse/DATAJPA-1357
+    // The issue is that these methods are called from the build result notification of bamboo, so there is no authentication object available.
+    // This doesn't cause problems for out-of-the-box repository methods, but does for custom ones.
+    private Authentication getAuthDummy() {
```

Comment:
do we still need this? If yes, I guess we have it in another place as well. Can we somehow avoid duplicating it?

Reply:
Is it also in another place? If so, I'm not aware.
This is a special case where there is no principal because Artemis is notified by Bamboo with a special token about the new build. It seems to be a known bug in Spring Boot that they want to fix (they should not check the principal because it can be null.

**I'll check though if this would be solved by using a std. jpa query.**

Reply:
Also in SecurityUtils.java.

Reply:
I migrated it to native jpa queries, so I think it won't be needed. I'll verify this on the test server.